File: 19.Backend Scaling and Performance Engineering - Part-2/code/typescript/example_1.ts
import Redis from 'ioredis';

const rdb = new Redis({ host: 'redis', port: 6379 });

export async function storeSession(sessionID: string, userID: string) {
    await rdb.set("session:" + sessionID, userID, 'EX', 24 * 60 * 60);
}

export async function getSession(sessionID: string) {
    return await rdb.get("session:" + sessionID);
}

File: 19.Backend Scaling and Performance Engineering - Part-2/code/rust/example_1.rs
use redis::AsyncCommands;

pub async fn store_session(mut con: redis::aio::Connection, session_id: &str, user_id: &str) -> redis::RedisResult<()> {
    con.set_ex(format!("session:{}", session_id), user_id, 24 * 3600).await
}

pub async fn get_session(mut con: redis::aio::Connection, session_id: &str) -> redis::RedisResult<String> {
    con.get(format!("session:{}", session_id)).await
}
